checks/raw/branch_protection.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raw
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/ossf/scorecard/v3/checker"
+	"github.com/ossf/scorecard/v3/clients"
+	sce "github.com/ossf/scorecard/v3/errors"
+)
+
+type branchMap map[string]*clients.BranchRef
+
+// BranchProtection retrieves the raw data for the Branch-Protection check.
+func BranchProtection(c clients.RepoClient) (checker.BranchProtectionsData, error) {
```

Comment:
same code as before but storing information in the raw structure. Nothing new really.